lib/msal-browser/src/interaction_handler/RedirectHandler.ts

```diff
@@ -51,29 +56,12 @@ export class RedirectHandler extends InteractionHandler {
             throw BrowserAuthError.createEmptyHashError(locationHash);
         }
 
-        // If navigateToLoginRequestUrl is true, then cache the hash and navigate to cached request URI.
-        if (this.navigateToLoginRequestUrl) {
-            this.browserStorage.setItem(TemporaryCacheKeys.URL_HASH, locationHash);
-            if (window.parent === window) {
-                const loginRequestUrl = this.browserStorage.getItem(TemporaryCacheKeys.ORIGIN_URI);
-
-                // Redirect to home page if login request url is null (real null or the string null)
-                if (!loginRequestUrl || loginRequestUrl === "null") {
-                    this.authModule.logger.error("Unable to get valid login request url from cache, redirecting to home page");
-                    window.location.href = "/";
-                } else {
-                    window.location.href = loginRequestUrl;
-                }
-            }
-            return null;
-        } else {
-            window.location.hash = "";
-        }
-
         // Interaction is completed - remove interaction status.
         this.browserStorage.removeItem(BrowserConstants.INTERACTION_STATUS_KEY);
         // Handle code response.
         const codeResponse = this.authModule.handleFragmentResponse(locationHash);
+        // Hash was processed successfully - remove from cache
+        this.browserStorage.removeItem(TemporaryCacheKeys.URL_HASH);
```

Comment:
@pkanher617 In v1 we think there might be a race case with this and `handleRedirectCallback`. fyi @tnorling